### PR TITLE
[cli]: show help when invalid flag is supplied

### LINKED ISSUE #1046: [ENHANCEMENT]: Prettify err in case the supplied flag(s) is/are unknown
**Describe the bug**
Presently thrown err are simply logged in case of unknown flag, maybe we can present it nicely and show a message to use help flag for proper usage

**What is the current behavior?**
Stacktrace is printed in case of unknown flag.

**To Reproduce**
Invoke cli with an unknown flag

**Expected behavior**
concise message of supplying an unknown flag should be displayed. something like - 
image

**Screenshots**
image

**Please paste the results of `webpack-cli info` here, and mention other relevant information**

**Additional context**
Looking for feedback on this before starting work.

### PR DESCRIPTION
**Pull request by anshumanv**
Fixes #1046 

**What kind of change does this PR introduce?**
intercept flag parsing and if any invalid flag is present, print a neat err.

**Did you add tests for your changes?**
Will add

**If relevant, did you update the documentation?**
NA

**Summary**
* Output pretty message when invalid flag is supplied.

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

File: lib/bootstrap.js [hunk 1]
```diff
@@ -6,17 +6,33 @@ require('./utils/process-log');
 
 const isFlagPresent = (args, flag) => args.find(arg => [flag, `--${flag}`].includes(arg));
 const normalizeFlags = (args, cmd) => args.slice(2).filter(arg => arg.indexOf('--') < 0 && arg !== cmd.name && arg !== cmd.alias);
+const validFlags = [...core.map(obj => obj.name), ...commands.map(obj => obj.name)]
```
Comment (anshumanv) [inline lib/bootstrap.js]: I feel I should just take core into consideration

File: lib/bootstrap.js [hunk 2]
```diff
@@ -6,17 +6,33 @@ require('./utils/process-log');
...
 
 const isCommandUsed = commands =>
     commands.find(cmd => {
         return process.argv.includes(cmd.name) || process.argv.includes(cmd.alias);
     });
 
+const checkValidFlag = () => {
+    let status = true
+    process.argv.slice(2).forEach(flag => {
```
Comment (anshumanv) [inline lib/bootstrap.js]: please suggest if you have refactoring for this in mind

**Comment by anshumanv** [comment]
Looks like this atm - 

image

**Comment by anshumanv** [comment]
debugging

**Comment by anshumanv** [comment]
Sounds good even!

File: lib/bootstrap.js (hunk 2 above)
Comment (jamesgeorge007) [reply lib/bootstrap.js]:
```js
const checkFlagIsValid = () => {
  return !process.argv.some(flag => {
    return !validFlags.includes(flag);    
  });
};
```
This is what I have in mind (seems a bit confusing due to the negations).

Comment (jamesgeorge007) [reply lib/bootstrap.js]: I guess you could get rid of the `status` variable and directly return a boolean value.

**Comment by anshumanv** [comment]
Will get back on this in a bit. 👍

**Comment by anshumanv** [comment]
We already have an unknown flag log when the flag is supplied, are there any other requirements/enhancements that we can do to it? @evenstensberg @ematipico

**Comment by ematipico** [comment]
Hi @anshumanv sorry to coming back to you so late, I completely missed this PR!

About the message provided, could you print `webpack --help` instead? If you rebased your PR, I would be happy to review it and merge it!

**Comment by anshumanv** [comment]
Hey Emanuale,

> About the message provided, could you print webpack --help instead

Def sounds better to be, will rebase and change it in a while 👍

**Comment by ematipico** [comment]
Thank you!

**Pull request merged**